src/player.cpp

```diff
@@ -1967,8 +1967,6 @@ bool player::takeoff( item &it, std::list<item> *res )
                            _( "<npcname> takes off their %s." ),
                            takeoff_copy.tname() );
 
-    // TODO: Make this variable
-    mod_moves( -250 );
```

Comment:
Can you help me understand how you are compensating for removing this line? This function does not appear to invoke `obtain_cost` in any way, so this would appear to make `player::takeoff` a free (instant) action now.

Because this function is where the actual removal of clothing occurs (and the messaging stating it has occurred), I think it makes sense for the movement cost to be deducted here as well, but maybe I have misunderstood the flow of operations.

Reply:
I understand what you're saying here. It is unclear when and where the move costs are tallied up, which makes it hard to track the actual move costs. 

This function in handle_action.cpp at line 1270 looks to me like it's already handling the obtain cost as it evaluates the parameter to be passed into the `player::takeoff` function. 

```
static void takeoff()
{
    avatar &player_character = get_avatar();
    item_location loc = game_menus::inv::take_off( player_character );

    if( loc ) {
        player_character.takeoff( *loc.obtain( player_character ) );
    } else {
        add_msg( _( "Never mind." ) );
    }
}
```

It modifies moves of the character who owns the location of the item before it passes that location to `player::takeoff`.

Reply:
The `player::takeoff` function is also invoked from `activity_actor.cpp`, `advanced_inv.cpp`, `game.cpp`, `npc.cpp` and `npcmove.cpp`. None of these appear to use the `static void takeoff()` function from `handle_action.cpp`, nor do they invoke `obtain` like that function does. I'm concerned that those other uses of `player::takeoff` will now take zero moves.

Reply:
Hmm, That makes me want to put the obtain_cost function inside of `player::takeoff` to make sure it happens every time, but then I'll have to de-tangle it from from the `handle_action.cpp` call where it happens "on the way."